src/xmms2-mpris.c

```diff
@@ -190,11 +190,26 @@ void handle_track_info(XmmsTrackInfo* info) {
         }
     }
 
+    // Fill Artist from title.
+    // Web radios mostly use title only: "Artist - Title" or "Artist / Title"
+    char **webtitle_array = NULL;
+    if (!(info->artist && info->artist[0] != '\0')) {
+        webtitle_array = g_strsplit(info->title, " - ", 2);
+        if (g_strv_length(webtitle_array) < 2) {
+            webtitle_array = g_strsplit(info->title, " / ", 2);
+        }
+        if (g_strv_length(webtitle_array) >= 2) {
+            info->artist = webtitle_array[0];
+            info->title = webtitle_array[1];
+        }
+    }
```

Comment:
```suggestion

    if (!(info->artist && info->artist[0] != '\0')) {
        webtitle_array = g_strsplit(info->title, " - ", 2);

        if (g_strv_length(webtitle_array) < 2) {
            webtitle_array = g_strsplit(info->title, " / ", 2);
        }

        if (g_strv_length(webtitle_array) >= 2) {
            info->artist = webtitle_array[0];
            info->title = webtitle_array[1];
        }
    }
```

This is good. The only change I ask for is just a bit more whitespace around blocks, which I'll add now.